### PR TITLE
[Refactor] Adapt to the new shadow partition

### PR DESCRIPTION
**Pull request by meegoo**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by meegoo** [comment]
run all

**Comment by meegoo** [comment]
run all

File: fe/fe-core/src/main/java/com/starrocks/planner/OlapTableSink.java [hunk 1]
```diff
@@ -415,6 +415,10 @@ private void setRangeKeys(RangePartitionInfo rangePartitionInfo, Partition parti
                         range.upperEndpoint().getKeys().get(i).treeToThrift().getNodes().get(0));
             }
         }
+
+        if (partition.getName().startsWith(ExpressionRangePartitionInfo.SHADOW_PARTITION_PREFIX)) {
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/planner/OlapTableSink.java:419]: What's the relationship between SHADOW PARTITION and AUTOMATIC PARTITION?

**Comment by meegoo** [comment]
run all

File: fe/fe-core/src/main/java/com/starrocks/planner/OlapTableSink.java (hunk 1 above)
Comment (Astralidea) [reply fe/fe-core/src/main/java/com/starrocks/planner/OlapTableSink.java:419]: AUTOMATIC PARTITION must have a SHADOW PARTITION, this is to ensure that this table has at least one tablet

**Comment by wanpengfei-git** [comment]
run all

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 8 / 10 (80.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|1|2|50.00%|[420]|
|:large_blue_circle:|com/starrocks/catalog/PartitionKey.java|7|8|87.50%|[102]|

**Review by Astralidea — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by meegoo** [comment]
run all

**Comment by wanpengfei-git** [comment]
run all

**Comment by wanpengfei-git** [comment]
run all

**Comment by meegoo** [comment]
run all

**Comment by meegoo** [comment]
run starrocks_admit_test

**Comment by meegoo** [comment]
run starrocks_be_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 8 / 10 (80.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|1|2|50.00%|[420]|
|:large_blue_circle:|com/starrocks/catalog/PartitionKey.java|7|8|87.50%|[102]|

**Comment by meegoo** [comment]
run starrocks_be_unittest

**Comment by meegoo** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by Astralidea — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by meegoo** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by meegoo** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 8 / 10 (80.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|1|2|50.00%|[420]|
|:large_blue_circle:|com/starrocks/catalog/PartitionKey.java|7|8|87.50%|[102]|

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**